Cannot Use Proxy That Returns Responses That Start With "HTTP/1.0"

- **OS and version used:** Windows 10

- **Java runtime used:** JRE 1.8.0_241

- **SDK version used:** IoT Device Client 1.18.0


# Description of the issue:
We are trying to use a proxy that returns its responses in HTTP 1.0. However, IoT SDK is only checking for responses that start with the string "HTTP/1.1", causing exceptions to be thrown.

Is it possible for the IoT Device Client library to support these kinds of proxies?

# Code sample exhibiting the issue:
```java
String azureConnectionString = "ourConnectionString";

client = new DeviceClient(azureConnectionString, IotHubClientProtocol.MQTT_WS);
RetryPolicy policy = new ExponentialBackoffWithJitter(RETRY_COUNT, RETRY_MIN_BACKOFF, RETRY_MAX_BACKOFF, RETRY_DELTA_BACKOFF, true);
client.setOperationTimeout(OPERATION_TIMEOUT);
client.setRetryPolicy(policy);

/* 
These two lines retrieve the proxy information
The proxy for this issue has no authentication information.
*/
Configuration configuration = Configuration.build();
Optional<ProxySettings> proxySettings = getProxySettingsFrom(configuration);

proxySettings.ifPresent(settings -> client.setProxySettings(settings));

client.open(); // This line throws the exceptions
```

# Console log of the issue:
```java
2020-02-25 15:22:14,680 [Thread-9] DEBUG com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection  - Opening MQTT connection...
2020-02-25 15:22:14,785 [Thread-9] DEBUG com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt  - Sending MQTT CONNECT packet...
2020-02-25 15:22:14,795 [MQTT Con: 4dc22e9c-0893-4630-bfee-7cdf3759f284] DEBUG com.microsoft.azure.sdk.iot.device.transport.ProxiedSSLSocket  - Sending tunnel handshake to HTTP proxy
2020-02-25 15:22:14,829 [Thread-9] WARN  com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt  - Exception encountered while sending MQTT CONNECT packet
MqttException (0) - java.io.IOException: Unable to tunnel through companyname.azure-devices.net:443.  Expected first response line to start with HTTP/1.1, but proxy returns "HTTP/1.0 200 Connection established"
	at org.eclipse.paho.client.mqttv3.internal.ExceptionHelper.createMqttException(ExceptionHelper.java:38)
	at org.eclipse.paho.client.mqttv3.internal.ClientComms$ConnectBG.run(ClientComms.java:736)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Unable to tunnel through companyname.azure-devices.net:443.  Expected first response line to start with HTTP/1.1, but proxy returns "HTTP/1.0 200 Connection established"
	at com.microsoft.azure.sdk.iot.device.transport.ProxiedSSLSocket.doTunnelHandshake(ProxiedSSLSocket.java:128)
	at com.microsoft.azure.sdk.iot.device.transport.ProxiedSSLSocket.connect(ProxiedSSLSocket.java:67)
	at org.eclipse.paho.client.mqttv3.internal.TCPNetworkModule.start(TCPNetworkModule.java:74)
	at org.eclipse.paho.client.mqttv3.internal.SSLNetworkModule.start(SSLNetworkModule.java:130)
	at org.eclipse.paho.client.mqttv3.internal.websocket.WebSocketSecureNetworkModule.start(WebSocketSecureNetworkModule.java:63)
	at org.eclipse.paho.client.mqttv3.internal.ClientComms$ConnectBG.run(ClientComms.java:722)
	... 1 more
2020-02-25 15:22:14,834 [Thread-9] DEBUG com.companyname.software.web.WebServer  - doStart() exception=
java.io.IOException: Could not open the connection
	at com.microsoft.azure.sdk.iot.device.DeviceIO.open(DeviceIO.java:162)
	at com.microsoft.azure.sdk.iot.device.InternalClient.open(InternalClient.java:123)
	at com.microsoft.azure.sdk.iot.device.DeviceClient.open(DeviceClient.java:316)
	at com.companyname.software.azure.AzureConnection.initialize(AzureConnection.java:184)
	at com.companyname.software.web.WebServer.lambda$doStart$0(WebServer.java:309)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.microsoft.azure.sdk.iot.device.exceptions.ProtocolException: Unable to establish MQTT connection
	at com.microsoft.azure.sdk.iot.device.transport.mqtt.exceptions.PahoExceptionTranslator.convertToMqttException(PahoExceptionTranslator.java:47)
	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.connect(Mqtt.java:130)
	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttMessaging.start(MqttMessaging.java:57)
	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.open(MqttIotHubConnection.java:187)
	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.openConnection(IotHubTransport.java:707)
	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.open(IotHubTransport.java:273)
	at com.microsoft.azure.sdk.iot.device.DeviceIO.open(DeviceIO.java:158)
	... 5 more
Caused by: MqttException (0) - java.io.IOException: Unable to tunnel through companyname.azure-devices.net:443.  Expected first response line to start with HTTP/1.1, but proxy returns "HTTP/1.0 200 Connection established"
	at org.eclipse.paho.client.mqttv3.internal.ExceptionHelper.createMqttException(ExceptionHelper.java:38)
	at org.eclipse.paho.client.mqttv3.internal.ClientComms$ConnectBG.run(ClientComms.java:736)
	... 1 more
Caused by: java.io.IOException: Unable to tunnel through companyname.azure-devices.net:443.  Expected first response line to start with HTTP/1.1, but proxy returns "HTTP/1.0 200 Connection established"
	at com.microsoft.azure.sdk.iot.device.transport.ProxiedSSLSocket.doTunnelHandshake(ProxiedSSLSocket.java:128)
	at com.microsoft.azure.sdk.iot.device.transport.ProxiedSSLSocket.connect(ProxiedSSLSocket.java:67)
	at org.eclipse.paho.client.mqttv3.internal.TCPNetworkModule.start(TCPNetworkModule.java:74)
	at org.eclipse.paho.client.mqttv3.internal.SSLNetworkModule.start(SSLNetworkModule.java:130)
	at org.eclipse.paho.client.mqttv3.internal.websocket.WebSocketSecureNetworkModule.start(WebSocketSecureNetworkModule.java:63)
	at org.eclipse.paho.client.mqttv3.internal.ClientComms$ConnectBG.run(ClientComms.java:722)
	... 1 more
```

--- a/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/ProxiedSSLSocket.java
+++ b/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/ProxiedSSLSocket.java
@@ -43,7 +43,9 @@ public class ProxiedSSLSocket extends SSLSocket
     private String proxyUsername;
     private char[] proxyPassword;
 
-    private static final String HTTP_VERSION_1_1 = "HTTP/1.1";
+    private static final String HTTP = "HTTP/";
+    private static final String HTTP_VERSION_1_1 = HTTP + "1.1";
+
 
     protected ProxiedSSLSocket(SSLSocketFactory socketFactory, Socket proxySocket, String proxyUsername, char[] proxyPassword)
     {
@@ -122,10 +124,10 @@ public class ProxiedSSLSocket extends SSLSocket
 
         //Expects the same http version in the response as the request
         String firstLine = connectResponseLines[connectResponseStart];
-        if (!firstLine.startsWith(HTTP_VERSION_1_1))
+        if (!firstLine.startsWith(HTTP))
         {
             tunnel.close();
-            throw new IOException(String.format("Unable to tunnel through %s:%d.  Expected first response line to start with %s, but proxy returns \"%s\"", host, port, HTTP_VERSION_1_1, firstLine));
+            throw new IOException(String.format("Unable to tunnel through %s:%d.  Expected first response line to start with %s, but proxy returns \"%s\"", host, port, HTTP, firstLine));
         }
 
         String[] replyStrParts = firstLine.split(" ");

